src/pages/iou/MoneyRequestSelectorPage.js

```diff
@@ -69,6 +70,18 @@ function MoneyRequestSelectorPage(props) {
         IOU.resetMoneyRequestInfo(moneyRequestID);
     };
 
+    const prevSelectedTab = usePrevious(props.selectedTab);
+
+    useEffect(() => {
+        if (prevSelectedTab === props.selectedTab) {
+            return;
+        }
+
+        resetMoneyRequestInfo();
+        // resetMoneyRequestInfo function is not added as dependencies since they don't change between renders
```

Comment:
Is this a necessary optimization?

Reply:
I asked whether it was a _necessary_ optimization. Disabling the exhaustiveness check for this dependency array is forever, for this dependency and for other potential future dependencies.

In other words, does it do any harm to include it in the deps? As I see it, you didn't answer this question.

Reply:
First, I'm trying to ensure that we're in agreement on whether this is an optimization that _can_ be done or something that _had to_ be done because otherwise the app behavior would be incorrect.

As I understand it, this the first case.

After giving it a thought, you might be right that silencing the lint might be acceptable here. It's a pity one can't disable this exhaustiveness check more granually, like (pseudo-code):

```
// eslint-config react-hooks/exhaustive-deps {resetMoneyRequestInfo: false}
```